python/pyarrow/tests/test_udf.py

```diff
@@ -0,0 +1,484 @@
+    def output_check(ctx, array):
+        ar = pc.call_function("add", [array, 1])
+        ar = ar.cast(pa.int32())
+        return ar
+    func_name = "test_output_value"
+    in_types = {"array": pa.int64()}
+    out_type = pa.int64()
+    doc = {
+        "summary": "add function scalar",
+        "description": "add function"
+    }
+    pc.register_scalar_function(output_check, func_name, doc,
+                                in_types, out_type)
+    return output_check, func_name
+
+
+@pytest.fixture(scope="session")
+def nullary_check_func_fixture():
+    # this needs to return array values
+    def nullary_check(ctx):
+        rand_vals = []
+        print("batch_length: ", ctx.batch_length)
+        for _ in range(ctx.batch_length):
+            random.seed(10)
+            rand_vals.append(random.randint(0, 10))
+        return pa.array(rand_vals)
+
+    func_doc = {
+        "summary": "random function",
+        "description": "generates a random value"
+    }
+    func_name = "test_random_func"
+    pc.register_scalar_function(nullary_check,
+                                func_name,
+                                func_doc,
+                                {},
+                                pa.int64())
+
+    return nullary_check, func_name
+
+
+def add_const(ctx, scalar):
+    return pc.call_function("add", [scalar, 1])
+
+
+@pytest.fixture(scope="session")
+def output_type_func_fixture():
+    func_name = "test_output_type"
+    in_types = {"array": pa.int64()}
+    out_type = pa.int64()
+    doc = {
+        "summary": "add function scalar",
+        "description": "add function"
+    }
+    pc.register_scalar_function(const_return, func_name, doc,
+                                in_types, out_type)
+    return const_return, func_name
+
+
+@pytest.fixture(scope="session")
+def varargs_check_func_fixture():
+    def varargs_check(ctx, *values):
+        base_val = values[:2]
+        res = pc.call_function("add", base_val)
+        for other_val in values[2:]:
+            res = pc.call_function("add", [res, other_val])
+        return res
+    func_name = "test_varargs_function"
+    in_types = {"array1": pa.int64(),
+                "array2": pa.int64(),
+                }
+    doc = {"summary": "n add function",
+           "description": "add N number of arrays"
+           }
+    pc.register_scalar_function(varargs_check, func_name, doc,
+                                in_types, pa.int64())
+
+    return varargs_check, func_name
+
+
+@pytest.fixture(scope="session")
+def raise_func_fixture():
+    def raise_func(ctx):
+        raise ValueError("Test function with raise")
+    func_name = "test_raise"
+    doc = {
+        "summary": "test function with raise",
+        "description": "function with a raise"
+    }
+    pc.register_scalar_function(raise_func, func_name, doc,
+                                {}, pa.int64())
+    return raise_func, func_name
+
+
+# TODO: just add the fixtures to this component
```

Comment:
Do you plan to address this here? Though I'm not sure what it means to do

Reply:
Ah nah, this was resolved in a better way. I think this comment no longer applies.